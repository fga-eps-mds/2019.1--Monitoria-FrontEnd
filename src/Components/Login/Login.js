import React, { Component } from 'react';
import './Login.css';
import { Grid, Button, TextField,} from '@material-ui/core' ;
import logo from '../../Assets/img/Logo.png';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import lightBlue from '@material-ui/core/colors/lightBlue';
import  {Link}  from 'react-router-dom';
import firebase from 'firebase';
import Typography from '@material-ui/core/Typography';
import CustomizedSnackbars from '../SimpleModal/Snackbars';

const theme = createMuiTheme({
  palette: {
    primary: { main: lightBlue[50] },
    secondary: { main: '#f44336' },
  },
  typography: { useNextVariants: true },
});

firebase.initializeApp({
  apiKey: process.env.REACT_APP_PATH_KEY,
  authDomain: process.env.REACT_APP_AUTH
});

class Login extends Component {
  state = {
    email: '',
    password: '',
    isAuthenticated: false,
    error: "",
    showError: false,
  };
  

  login = async (e) => {
    
    const { email, password } = this.state;
    if(!email || !password){
      this.setState({ error: "Preencha e-mail e senha para continuar!" });
      this.setState({ showError: true});
      e.preventDefault();
    }else{
      await firebase.auth().signInWithEmailAndPassword(email, password).then((user)=>{
        this.setState({isAuthenticated: true});
      }).catch((except)=>{
        this.setState({ error: "Email ou Senha inv√°lidos." });
        this.setState({ showError: true});
      });
    }
  }

  render() {
    return (
      <div className="LoginBackground" style={{overflowY:'auto',overflowX:'hidden'}}>
        <Grid container alignContent="center" justify="center" direction="column" alignItems="center">
          
          <img src={logo} alt="Logo" />
          <Grid item >
            <TextField
              error = {this.state.error}
              id="emailTextField"
              label="Email"
              margin="normal"
              required= "true"
              value={this.state.email}
              onChange={(event)=>this.setState({
              email: event.target.value,
              })}
              />
          </Grid>
          <Grid item >
            <TextField
              error = {this.state.error}
              id="senhaTextField"
              label="Senha"
              margin="normal"
              type="password"
              required= "true"
              value={this.state.password}
              onChange={(event)=>this.setState({
              password: event.target.value,
              })}
              />
          </Grid>
          <Grid container alignContent="center" justify="center" direction="row" spacing={50} alignItems="center">
              {this.state.showError? <CustomizedSnackbars error={this.state.error}/>:null}
          </Grid>
        </Grid>
        <Grid  container alignContent="center" justify="center" direction="column" spacing={24} alignItems="center" style={{marginTop: 25}}>
          <Grid item >
            <MuiThemeProvider theme={theme}>
              <Button component={Link} to={this.state.isAuthenticated?"/Feed":"/"} variant="outlined" color="primary" onClick={this.login}>
                  Login
              </Button>
            </MuiThemeProvider>
          </Grid>  
          <Grid item>
            <MuiThemeProvider theme={theme}>
              <Button component={Link} to="/SignUp" variant="outlined" color="primary">
                Registrar
              </Button>
              </MuiThemeProvider>
          </Grid>  
              <Link to="/ForgotPassword" style={{color:'#bdbdbd'}} >Esqueceu sua senha ?</Link>
        </Grid>
          
        </div>
    );
  }
}

export default Login;
