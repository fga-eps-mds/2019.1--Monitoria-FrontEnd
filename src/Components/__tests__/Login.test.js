import React from 'react';
import { shallow } from 'enzyme';
import Login from '../Login/Login';
import toJson from 'enzyme-to-json';

describe('Testing Login component', () => {
    it('Test if Login renders correctly', () =>{
        const tree = shallow(<Login/>);
        expect(toJson(tree)).toMatchSnapshot();
    });
});
