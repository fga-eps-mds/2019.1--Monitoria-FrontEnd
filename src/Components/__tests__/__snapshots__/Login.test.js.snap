// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login component Test if Login renders correctly 1`] = `
<Route>
  <Component />
</Route>
`;
